lib/tailreadstream/tailreadstream.js

```diff
@@ -217,7 +226,8 @@ class TailReadStream extends Readable {
 
     fs.stat(this._filepath, (err, stats) => {
       if (err) {
-        return setTimeout(that._watchFile.bind(that), that._watchinterval)
+        this._watchTimer = setTimeout(that._watchFile.bind(that), that._watchinterval)
```

Comment:
do we need a check for `this._exiting`? seems like there could be timing issues with how/when `_watchFile` and `destroy` get called, but not totally sure.

Reply:
There's a small window of opportunity for a race condition, sure.  Couldn't hurt to add it.